File: frontend/src/api/map.ts
/**
 * API calls untuk modul Map/Heatmap.
 */
import { apiClient } from "./client";

export interface DiseasePoint {
  scan_id: string;
  lat: number;
  lng: number;
  disease: string;
  crop_type: "rice" | "corn";
  confidence: number;
  month: string;
}

export interface HeatmapData {
  points: DiseasePoint[];
  total: number;
}

export interface HeatmapParams {
  crop_type?: "rice" | "corn";
  disease?: string;
  months?: number;
}

export const mapApi = {
  getHeatmap: async (params: HeatmapParams = {}): Promise<HeatmapData> => {
    const { data } = await apiClient.get<{ success: boolean; data: HeatmapData }>(
      "/map/heatmap",
      { params }
    );
    return data.data;
  },
};

File: backend/app/schemas/map.py
"""Pydantic schemas untuk modul Map/Heatmap."""

from pydantic import BaseModel


class DiseasePoint(BaseModel):
    """Satu titik penyakit — scan yang sudah selesai dan punya koordinat."""
    scan_id: str
    lat: float
    lng: float
    disease: str
    crop_type: str
    confidence: float
    # Format "YYYY-MM" untuk grouping timeline
    month: str


class HeatmapResponse(BaseModel):
    points: list[DiseasePoint]
    total: int
